### PR TITLE
Implements BOM tests for generator

### PR DESCRIPTION
**Pull request by Karm**
All, including productized Quarkus builds nad native images:

```
mvn clean verify -Ptestsuite 
 -Dquarkus.version=1.3.0.Final-redhat-00019
 -Dquarkus.platform.version=1.3.0.Final-redhat-00010
 -Dtests.maven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository
 -Dmaven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository
```

Community version, including native images:

```
mvn clean verify -Ptestsuite-community -Dquarkus.version=1.3.0.Final
```

### CODE REVIEW

**Comment by Karm** [comment]
Centos slave is down and I cannot reach it now :-(

File: README.md [hunk 1]
```diff
@@ -3,6 +3,23 @@
...
 
+
+All, including Quarkus builds nad native images:
+
+```
+mvn clean verify -Ptestsuite 
+ -Dquarkus.version=1.3.0.Final-redhat-00019
+ -Dquarkus.platform.version=1.3.0.Final-redhat-00010
+ -Dtests.maven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository
+ -Dmaven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository
```
Comment (rsvoboda) [inline README.md]:
```suggestion
mvn clean verify -Ptestsuite \
 -Dquarkus.version=1.3.0.Final-redhat-00019 \
 -Dquarkus.platform.version=1.3.0.Final-redhat-00010 \
 -Dtests.maven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository \
 -Dmaven.repo.local=/home/karm/QUARKUS/quarkus-1.3.0.ER11/maven-repository
```

File: app-generated-skeleton/application.properties [hunk 2]
```diff
@@ -1,23 +1,17 @@
 # This is a FAKE configuration to boot the generated empty project.
 # No code actually uses it at runtime to show any examples/quickstarts.
 
+quarkus.oidc.auth-server-url=http://localhost:6661/auth/realms/quarkus
```
Comment (rsvoboda) [inline app-generated-skeleton/application.properties]: +1 to create question/issue about eager requirement of running keycloak server

File: testsuite/src/it/java/io/quarkus/ts/startstop/ArtifactGeneratorTest.java [hunk 3]
```diff
@@ -69,7 +73,71 @@
 
     private static final Logger LOGGER = Logger.getLogger(ArtifactGeneratorTest.class.getName());
 
-    public void testRuntime(TestInfo testInfo, String[] extensions, boolean warmUp) throws IOException, InterruptedException {
+    public static final String[] allSupportedExtensionsSetA = new String[]{
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/ArtifactGeneratorTest.java]: minor: allSupportedExtensionsSetA => supportedExtensionsSetA

same for SetB

File: testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * Whitelists jar names.
+ *
+ * There are basically known issues. The enum should be empty.
+ *
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+public enum WhitelistProductBomJars {
+    PRODUCT_BOM(new String[]{
+            "lib/jakarta.",
+            "lib/javax.",
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java]: these shouldn't be there

File: testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            "lib/com.google.code.findbugs.jsr305",
+            "lib/com.google.guava.failureaccess",
+            "lib/org.eclipse.yasson",
+            "lib/com.mchange.mchange-commons-java",
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java]: no longer issue in ER11

File: testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            "lib/org.checkerframework.checker-qual",
+            "lib/io.smallrye.reactive.smallrye-axle-amqp-client"
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/utils/WhitelistProductBomJars.java]: no longer issue in ER11

I think only `org.checkerframework.checker-qual` should stay there

File: pom.xml [hunk 7]
```diff
@@ -31,8 +31,7 @@
     <name>Quarkus StartStop TS: Parent</name>
 
     <properties>
-        <!--<quarkus.version>0.0.1582091968553.temporary-redhat-00005</quarkus.version>-->
-        <quarkus.version>1.3.0.Alpha2</quarkus.version>
+        <quarkus.version>1.3.0.Final</quarkus.version>
```
Comment (rsvoboda) [inline pom.xml]: Please bump it to 1.3.1.Final

**Comment by Karm** [comment]
@rsvoboda  @mjurc  IMHO ready.

**Pull request merged**